<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SessionSettingsDialog</class>
 <widget class="QDialog" name="SessionSettingsDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>460</width>
    <height>580</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Session Settings</string>
  </property>
  <layout class="QVBoxLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="settingsTab">
      <attribute name="title">
       <string>Settings</string>
      </attribute>
      <layout class="QGridLayout">
       <item row="4" column="0" colspan="3">
        <widget class="QGroupBox" name="sessionGroup">
         <property name="title">
          <string>Session</string>
         </property>
         <layout class="QFormLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="opwordLabel">
            <property name="text">
             <string>Operator password:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="opword">
            <property name="text">
             <string notr="true">&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;not set&lt;/span&gt; (&lt;a href=&quot;#change&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#2980b9;&quot;&gt;assign&lt;/span&gt;&lt;/a&gt;)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="autoresetLabel">
            <property name="text">
             <string>Autoreset threshold:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <layout class="QHBoxLayout" name="autoresetLayout">
            <item>
             <widget class="QDoubleSpinBox" name="autoresetThreshold">
              <property name="toolTip">
               <string>Session history size at which autoreset triggers</string>
              </property>
              <property name="specialValueText">
               <string>No autoreset</string>
              </property>
              <property name="suffix">
               <string> MB</string>
              </property>
              <property name="decimals">
               <number>1</number>
              </property>
              <property name="minimum">
               <double>0.000000000000000</double>
              </property>
              <property name="value">
               <double>0.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="baseResetThreshold">
              <property name="text">
               <string notr="true">+ 0 MB</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="sessionOptionsLabel">
            <property name="text">
             <string>Options:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QCheckBox" name="preserveChat">
            <property name="toolTip">
             <string>Include chat in session history so new users can see it as well</string>
            </property>
            <property name="text">
             <string>Keep chat history</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QCheckBox" name="persistent">
            <property name="toolTip">
             <string>Don't terminate session when the last user leaves</string>
            </property>
            <property name="text">
             <string>Persist without users</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QCheckBox" name="nsfm">
            <property name="toolTip">
             <string>This session contains age-restricted content</string>
            </property>
            <property name="text">
             <string>Not suitable for minors (NSFM)</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="idleTimeoutLabel">
            <property name="text">
             <string>Idle timeout:</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QLabel" name="idleTimeoutAmount">
            <property name="text">
             <string>X seconds</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QCheckBox" name="idleTimeoutOverride">
            <property name="text">
             <string>Disable idle timeout</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0" colspan="3">
        <widget class="QLineEdit" name="title"/>
       </item>
       <item row="0" column="0" colspan="3">
        <widget class="QLabel" name="titleLabel">
         <property name="text">
          <string>Title:</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <spacer name="settingsBottomSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="0" rowspan="2" colspan="3">
        <widget class="QGroupBox" name="loginGroup">
         <property name="title">
          <string>Login</string>
         </property>
         <layout class="QFormLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="sessionPasswordLabel">
            <property name="text">
             <string>Password:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="sessionPassword">
            <property name="text">
             <string notr="true">&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;not set&lt;/span&gt; (&lt;a href=&quot;#change&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#2980b9;&quot;&gt;assign&lt;/span&gt;&lt;/a&gt;)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="maxUsersLabel">
            <property name="text">
             <string>Max users:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="maxUsers">
            <property name="toolTip">
             <string>How many users to let in. Does not affect users already logged in</string>
            </property>
            <property name="minimum">
             <number>2</number>
            </property>
            <property name="maximum">
             <number>254</number>
            </property>
            <property name="value">
             <number>25</number>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <spacer name="loginSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>10</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QCheckBox" name="denyJoins">
            <property name="text">
             <string>Block new joins</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0" colspan="2">
           <widget class="QCheckBox" name="authOnly">
            <property name="text">
             <string>Require a registered account to join</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0" colspan="2">
           <widget class="QCheckBox" name="allowWeb">
            <property name="text">
             <string>Allow joining from the web browser</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="permissionsTab">
      <attribute name="title">
       <string>Permissions</string>
      </attribute>
      <layout class="QFormLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="labelPutImage">
         <property name="text">
          <string>Cut, Paste &amp; Fill:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QComboBox" name="permPutImage"/>
       </item>
       <item row="1" column="1">
        <widget class="QComboBox" name="permRegionMove"/>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="labelCreateAnnotation">
         <property name="text">
          <string>New Annotations:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QComboBox" name="permCreateAnnotation"/>
       </item>
       <item row="15" column="0" colspan="2">
        <widget class="widgets::PresetSelector" name="permissionPresets" native="true">
         <property name="presetFolder" stdset="0">
          <string>permissions</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="labelEditLayers">
         <property name="text">
          <string>General Layer Editing:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QComboBox" name="permEditLayers"/>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="labelOwnLayers">
         <property name="text">
          <string>Own Layer Editing:</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QComboBox" name="permOwnLayers"/>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="labelResize">
         <property name="text">
          <string>Canvas Resizing:</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QComboBox" name="permResize"/>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="labelBackground">
         <property name="text">
          <string>Canvas Background:</string>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="QComboBox" name="permBackground"/>
       </item>
       <item row="8" column="0">
        <widget class="QLabel" name="labelLaser">
         <property name="text">
          <string>Laser Pointer:</string>
         </property>
        </widget>
       </item>
       <item row="8" column="1">
        <widget class="QComboBox" name="permLaser"/>
       </item>
       <item row="9" column="0">
        <widget class="QLabel" name="labelUndo">
         <property name="text">
          <string>Undo &amp; Redo:</string>
         </property>
        </widget>
       </item>
       <item row="9" column="1">
        <widget class="QComboBox" name="permUndo"/>
       </item>
       <item row="10" column="0">
        <widget class="QLabel" name="labelDeputies">
         <property name="text">
          <string>Kick &amp; Ban:</string>
         </property>
        </widget>
       </item>
       <item row="10" column="1">
        <widget class="QComboBox" name="deputies">
         <item>
          <property name="text">
           <string>Operators</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Trusted</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="11" column="0">
        <widget class="QLabel" name="labelTimeline">
         <property name="text">
          <string>Timeline:</string>
         </property>
        </widget>
       </item>
       <item row="11" column="1">
        <widget class="QComboBox" name="permTimeline"/>
       </item>
       <item row="12" column="0">
        <widget class="QLabel" name="labelMetadata">
         <property name="text">
          <string>Metadata:</string>
         </property>
        </widget>
       </item>
       <item row="12" column="1">
        <widget class="QComboBox" name="permMetadata"/>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="labelRegionMove">
         <property name="text">
          <string>Selection Move &amp; Resize:</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="labelMyPaint">
         <property name="text">
          <string>MyPaint Brushes:</string>
         </property>
        </widget>
       </item>
       <item row="7" column="1">
        <widget class="QComboBox" name="permMyPaint"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="bansTab">
      <attribute name="title">
       <string>Bans</string>
      </attribute>
      <layout class="QVBoxLayout">
       <item>
        <widget class="QLabel" name="banLabel">
         <property name="text">
          <string>Only operators can manage bans.</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTableView" name="banlistView">
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="bansButtonLayout">
         <item>
          <widget class="QPushButton" name="importButton">
           <property name="text">
            <string>Import</string>
           </property>
           <property name="icon">
            <iconset theme="document-import">
             <normaloff>theme:document-import.svg</normaloff>theme:document-import.svg</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="exportButton">
           <property name="text">
            <string>Export</string>
           </property>
           <property name="icon">
            <iconset theme="document-export">
             <normaloff>theme:document-export.svg</normaloff>theme:document-export.svg</iconset>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="bansButtonSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="removeBan">
           <property name="text">
            <string>Remove selected</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="authListPage">
      <attribute name="title">
       <string>Roles</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="authLabel">
         <property name="text">
          <string notr="true"/>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTableView" name="authListView">
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
         <property name="sortingEnabled">
          <bool>true</bool>
         </property>
         <attribute name="horizontalHeaderHighlightSections">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="authButtonLayout">
         <item>
          <widget class="QPushButton" name="authImportButton">
           <property name="text">
            <string>Import</string>
           </property>
           <property name="icon">
            <iconset theme="document-import">
             <normaloff>theme:document-import.svg</normaloff>theme:document-import.svg</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="authExportButton">
           <property name="text">
            <string>Export</string>
           </property>
           <property name="icon">
            <iconset theme="document-export">
             <normaloff>theme:document-export.svg</normaloff>theme:document-export.svg</iconset>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="authButtonSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QCheckBox" name="authOp">
           <property name="text">
            <string>Operator</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="authTrusted">
           <property name="text">
            <string>Trusted</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="announcementsTab">
      <attribute name="title">
       <string>Announcements</string>
      </attribute>
      <layout class="QVBoxLayout">
       <item>
        <widget class="QListView" name="announcementListView">
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="announcementsButtonLayout">
         <item>
          <spacer name="announcementsButtonSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="addAnnouncement">
           <property name="text">
            <string>Add listing</string>
           </property>
           <property name="icon">
            <iconset theme="list-add">
             <normaloff>theme:list-add.svg</normaloff>theme:list-add.svg</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="removeAnnouncement">
           <property name="text">
            <string>Remove selected listing</string>
           </property>
           <property name="icon">
            <iconset theme="list-remove">
             <normaloff>theme:list-remove.svg</normaloff>theme:list-remove.svg</iconset>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Close</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>widgets::PresetSelector</class>
   <extends>QWidget</extends>
   <header>desktop/widgets/presetselector.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>title</tabstop>
  <tabstop>maxUsers</tabstop>
  <tabstop>denyJoins</tabstop>
  <tabstop>authOnly</tabstop>
  <tabstop>autoresetThreshold</tabstop>
  <tabstop>preserveChat</tabstop>
  <tabstop>persistent</tabstop>
  <tabstop>nsfm</tabstop>
  <tabstop>permPutImage</tabstop>
  <tabstop>permRegionMove</tabstop>
  <tabstop>permCreateAnnotation</tabstop>
  <tabstop>permEditLayers</tabstop>
  <tabstop>permOwnLayers</tabstop>
  <tabstop>permResize</tabstop>
  <tabstop>permBackground</tabstop>
  <tabstop>permLaser</tabstop>
  <tabstop>permUndo</tabstop>
  <tabstop>deputies</tabstop>
  <tabstop>permTimeline</tabstop>
  <tabstop>permMetadata</tabstop>
  <tabstop>banlistView</tabstop>
  <tabstop>removeBan</tabstop>
  <tabstop>announcementListView</tabstop>
  <tabstop>addAnnouncement</tabstop>
  <tabstop>removeAnnouncement</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>SessionSettingsDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>224</x>
     <y>453</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>SessionSettingsDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>292</x>
     <y>459</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
